src/requirements/Farcaster/components/FarcasterCast.tsx

```diff
@@ -0,0 +1,155 @@
+import {
+  HStack,
+  Icon,
+  Image,
+  Link,
+  MergeWithAs,
+  Spinner,
+  Stack,
+  StackProps,
+  Text,
+  useColorModeValue,
+} from "@chakra-ui/react"
+import { CaretRight, Chat, Heart, ShareNetwork, WarningCircle } from "phosphor-react"
+import { PropsWithChildren } from "react"
+import FarcasterCastSmall from "./FarcasterCastSmall"
+
+export type FarcasterCastData = {
+  hash: string
+  username: string
+  display_name: string
+  profile_pic: string
+  text: string
+  timestamp: string
+  likes: number
+  recasts: number
+  replies: number
+}
+
+const FarcasterCast = ({
+  cast,
+  loading,
+  error,
+  size = "md",
+}: {
+  cast: FarcasterCastData
+  loading: boolean
+  error: boolean
+  size?: string
+}) => {
+  const bg = useColorModeValue("gray.50", "blackAlpha.200")
+  const bgHover = useColorModeValue("gray.100", "blackAlpha.300")
+
+  const url = `https://warpcast.com/${cast?.username}/${cast?.hash}`
+  const prettyDate =
+    cast?.timestamp &&
+    new Intl.DateTimeFormat("en-US", {
+      dateStyle: "short",
+      timeStyle: "short",
+    }).format(new Date(cast.timestamp))
+
+  if (size === "sm") {
+    return <FarcasterCastSmall cast={cast} error={error} loading={loading} />
+  }
+
+  if (loading) {
+    return (
+      <CastWrapper>
+        <Spinner mr={2} size="sm" />
+        <Text fontSize="sm">Loading cast...</Text>
+      </CastWrapper>
+    )
+  }
+
+  if (error) {
+    return (
+      <CastWrapper>
+        <Icon as={WarningCircle} />
+        <Text fontSize="sm" textAlign="center">
+          Failed to load cast!
+        </Text>
+      </CastWrapper>
+    )
+  }
+
+  if (!cast) {
+    return (
+      <CastWrapper>
+        <Text fontSize="sm" textAlign="center" opacity={0.6}>
+          No cast found
+        </Text>
+      </CastWrapper>
+    )
+  }
+
+  return (
+    <CastWrapper
+      as={Link}
+      href={url}
+      isExternal
+      _hover={{ bg: bgHover, cursor: "pointer", textDecoration: "none" }}
+      justifyContent="start"
+      spacing={4}
+    >
+      <Image
+        width={7}
+        height={7}
+        objectFit="cover"
+        rounded="full"
+        src={cast.profile_pic}
+        alt="Profile picture"
+      />
+      <Stack spacing={0}>
+        <Text fontWeight="bold" fontSize="sm" noOfLines={1}>
+          {cast.display_name}
+        </Text>
+        <Text fontSize="xs" opacity={0.6}>
+          {prettyDate}
+        </Text>
+      </Stack>
+      <HStack spacing={3} ml="auto">
+        <Stat icon={Heart} value={cast.likes} />
+        <Stat icon={ShareNetwork} value={cast.recasts} />
+        <Stat icon={Chat} value={cast.replies} />
+      </HStack>
+      <Icon as={CaretRight} />
```

Comment:
We should use ArrowSquareOut here imo, that expresses better that it's an external link and we follow that pattern in the app (also CaretRight for same-page navigation, and ArrowSquareIn for modals - this latter one is not ideal tho)